src/core/assert/base.js

```diff
@@ -57,18 +57,13 @@ export function assert(
     opt_message += sentinel;
   }
 
-  // Skip the first 3 arguments to isolate format params
-  // const messageArgs = Array.prototype.slice.call(arguments, 3);
-  // Index at which message args start
-  let i = 3;
-
   // Substitute provided values into format string in message
   const splitMessage = opt_message.split('%s');
   let message = splitMessage.shift();
   const messageArray = [message];
 
   while (splitMessage.length) {
-    const subValue = arguments[i++];
+    const subValue = msgArgs.shift();
```

Comment:
This turns an O(n) into O(n<sup>2</sup>) operation.

Reply:
Reverting all changes to this file